### PR TITLE
Origin/scheduler config tests

### LINKED ISSUE #1805: Add Unit and Integration tests for scheduler services
**There are no unit tests in place for testing the functionalities of the scheduler**

Unit tests for services within the scheduler package, most notably SchedulerConfig, should be added to make sure that scheduling is executed as expected, edge cases are covered, and exceptions are handled gracefully.

### PR DESCRIPTION
**Pull request by OsmanRahmoun**
# Pull Requests Requirements

- [ ] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [ ] The changes confirm to the OpenElis Global x3 Styleguide and design
      documentation.
- [ ] The changes include tests or are validated by existing tests.
- [ ] I have read and agree to the Contributing Guidelines of this project.

## Summary

Expanding on contributions made by @Willpon4, I have expanded testing of SchedulerConfig to include:

**testTaskExecutor**: This test calls the method that creates a task executor. It verifies that the returned Executor is not null.

**testReloadSchedulesWithEmptyList**: simulates a situation where there are no cron schedules. The scheduler is reloaded, and the test checks that the scheduler is shut down and that no jobs are scheduled.

**testReloadSchedulesWithCronStatementNever**: This test creates a cron schedule with the value "never", which means it should not run. The scheduler is reloaded, and the test makes sure that the scheduler shuts down and does not schedule any jobs.

## Screenshots

[Add relevant screenshots here if applicable]

## Related Issue

Resolves #1805 

## Other

[Add any additional information or notes here]

### CODE REVIEW

File: src/test/java/org/openelisglobal/scheduler/SchedulerConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import static org.mockito.Mockito.*;
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
```
Comment (Agaba-derrick) [inline src/test/java/org/openelisglobal/scheduler/SchedulerConfigTest.java:13]: try to avoid injecting mocks in your tests instead use   datasets

File: src/test/java/org/openelisglobal/scheduler/SchedulerConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        long result = schedulerConfig.getResultsResendTimeMillis();
+
+        assertEquals(default_period_millis, result);
+    }
+
+    @Test
+    public void testTaskExecutor() {
+        // When
+        Executor executor = schedulerConfig.taskExecutor();
+        // Then
```
Comment (Agaba-derrick) [inline src/test/java/org/openelisglobal/scheduler/SchedulerConfigTest.java:58]: avoid un necesary comments in our code !

**Review by Agaba-derrick — CHANGES_REQUESTED** [review]

**Comment by Agaba-derrick** [comment]
please add  screen shot showing the current changes   you made

File: src/test/java/org/openelisglobal/scheduler/SchedulerConfigTest.java (hunk 1 above)
Comment (OsmanRahmoun) [reply src/test/java/org/openelisglobal/scheduler/SchedulerConfigTest.java:13]: With testing SchedulerConfigService, I notice that utilizing datasets may not be possible in testing the functionalities, and mocking is necessary as it tests how objects behave. Can you confirm this @Agaba-derrick . If not, can we connect to maybe show me what you mean?

**Comment by mozzy11** [comment]
Thanks @OsmanRahmoun
